Android editor build error in etcpak on armv7 with NDK r23

### Godot version

4.0.dev (26dd4746a1a979132ef81c4674d3ea558a77710a)

### System information

Fedora 36, compiling for Android armv7 with NDK r23

### Issue description

Following #61691, it seems Android editor builds are failing in `master` due to an error in the etcpak dependency with some NEON intrinsics:

```
thirdparty/etcpak/ProcessRGB.cpp:3259:16: error: argument value 4 is outside the valid range 
        return vdupq_lane_s16( vget_low_s16( multipliers ), ClampConstant( Lane, 0, 4 ) );
               ^                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/linux-x86_64/lib64/clang/12.0.9/include/arm_neon.h:5979:14: note: expanded from macro 'vdupq_lane_s16'
  __ret_24 = splatq_lane_s16(__s0_24, __p1_24); \
             ^                        ~~~~~~~
/root/sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/linux-x86_64/lib64/clang/12.0.9/include/arm_neon.h:829:23: note: expanded from macro 'splatq_lane_s16'
  __ret = (int16x8_t) __builtin_neon_splatq_lane_v((int8x8_t)__s0, __p1, 1); \
                      ^                                            ~~~~
thirdparty/etcpak/ProcessRGB.cpp:3723:46: note: in instantiation of function template specialization 'WidenMultiplier_EAC_NEON<8>' requested here
    uint8x8_t recVals[16] = { EAC_APPLY_16X( EAC_RECONSTRUCT_VALUE ) };
                                             ^
```

I've had to fix similar issues in the past (https://github.com/wolfpld/etcpak/issues/25), seems like my fixes worked for NDK r21 but not for NDK r23 or some of our build config changes. CC @madmiraal 

### Steps to reproduce

- `scons p=android tools=yes target=release_debug android_arch=armv7`

### Minimal reproduction project

_No response_

--- a/thirdparty/README.md
+++ b/thirdparty/README.md
@@ -118,7 +118,7 @@ will limit its functionality to IPv4 only.
 ## etcpak
 
 - Upstream: https://github.com/wolfpld/etcpak
-- Version: 1.0 (a77d5a37ddf48034cee8aeb9e8792a623c265b4c, 2022)
+- Version: 1.0 (153f0e04a18b93c277684b577365210adcf8e11c, 2022)
 - License: BSD-3-Clause
 
 Files extracted from upstream source:

--- a/thirdparty/etcpak/ProcessRGB.cpp
+++ b/thirdparty/etcpak/ProcessRGB.cpp
@@ -3216,9 +3216,9 @@ etcpak_force_inline static uint16x8_t ErrorProbe_EAC_NEON( uint8x8_t recVal, uin
     uint8x8_t srcValWide;
 #ifndef __aarch64__
     if( Index < 8 )
-        srcValWide = vdup_lane_u8( vget_low_u8( alphaBlock ), ClampConstant( Index, 0, 8 ) );
+        srcValWide = vdup_lane_u8( vget_low_u8( alphaBlock ), ClampConstant( Index, 0, 7 ) );
     else
-        srcValWide = vdup_lane_u8( vget_high_u8( alphaBlock ), ClampConstant( Index - 8, 0, 8 ) );
+        srcValWide = vdup_lane_u8( vget_high_u8( alphaBlock ), ClampConstant( Index - 8, 0, 7 ) );
 #else
     srcValWide = vdup_laneq_u8( alphaBlock, Index );
 #endif
@@ -3256,9 +3256,9 @@ etcpak_force_inline static int16x8_t WidenMultiplier_EAC_NEON( int16x8_t multipl
     constexpr int Lane = GetMulSel( Index );
 #ifndef __aarch64__
     if( Lane < 4 )
-        return vdupq_lane_s16( vget_low_s16( multipliers ), ClampConstant( Lane, 0, 4 ) );
+        return vdupq_lane_s16( vget_low_s16( multipliers ), ClampConstant( Lane, 0, 3 ) );
     else
-        return vdupq_lane_s16( vget_high_s16( multipliers ), ClampConstant( Lane - 4, 0, 4 ) );
+        return vdupq_lane_s16( vget_high_s16( multipliers ), ClampConstant( Lane - 4, 0, 3 ) );
 #else
     return vdupq_laneq_s16( multipliers, Lane );
 #endif

